src/prefect/workers/base.py

```diff
@@ -502,15 +502,19 @@ def get_name_slug(self):
         return slugify(self.name)
 
     def get_flow_run_logger(self, flow_run: "FlowRun") -> PrefectLogAdapter:
+        extra = {
+            "worker_name": self.name,
+            "work_pool_name": (
+                self._work_pool_name if self._work_pool else "<unknown>"
+            ),
+            "work_pool_id": str(getattr(self._work_pool, "id", "unknown")),
+        }
+        if self.backend_id:
```

Comment:
While worker logs is experimental, we'll only have a backend ID here if the user has opted in to the feature and we're running against Cloud.